code/mission/missiontraining.cpp

```diff
@@ -658,7 +658,8 @@ SCP_string message_translate_tokens(const char *text)
 				ptr = translate_key(temp);  // try and translate key
 				if (ptr != nullptr) {  // was key translated properly?
 					if (!stricmp(ptr, NOX("none")) && (Training_bind_warning != Missiontime)) {
-						if ( The_mission.game_type & MISSION_TYPE_TRAINING ) {
+						// check if a warning message should be displayed if the key is unbound
+						if ( (The_mission.game_type & MISSION_TYPE_TRAINING) || (Always_warn_player_about_unbound_keys & MISSION_TYPE_SINGLE) ) {
 							r = popup(PF_TITLE_BIG | PF_TITLE_RED, 2, XSTR( "&Bind Control", 424), XSTR( "&Abort mission", 425),
 								XSTR( "Warning\nYou have no control bound to the action \"%s\".  You must do so before you can continue with your training.", 426),
```

Comment:
Super annoying, I know but if they're in a regular mission, it's probably weird to have it say, "continue with your training."  Is there a way to add a separate `XSTR` for regular single player missions?

Reply:
Yeah, that's something I was thinking about as well. There's a few options I was thinking of:
1) Modders can set their own string in tbm of the strings table.
2) Add a new string/XSTR, though I reckon that it should have the base translations added too(?)
3) Change the existing string in the default tables.

Reply:
Changing the existing string means we invalidate additional translations that have happened since retail, right?

Adding the new string with corresponding base translations seems to be the best option to me, even though it is more work.  But what do others think?

Reply:
`Adding the new string with corresponding base translations seems to be the best option to me, even though it is more work.` Yeah, I agree that's probably the best bet.

Reply:
So, after a quick chat on discord, we decided option 1 is better for now.